### PR TITLE
[prometheus-cloudwatch-exporter] Update default image tag for Cloudwatch exporter chart

### PR DESCRIPTION
**Pull request by duffn**
#### What this PR does / why we need it:

This updates the default tag for the Cloudwatch exporter from v0.12.2 to v0.14.0. This is particularly important because v0.13.3 fixed CVE-2022-21449, the "psychic signatures" vulnerability: https://nvd.nist.gov/vuln/detail/CVE-2022-21449

- Tag diff: https://github.com/prometheus/cloudwatch_exporter/compare/v0.12.2...v0.14.0
- Fix for psychic signatures vulnerability: https://github.com/prometheus/cloudwatch_exporter/issues/415

#### Which issue this PR fixes

n/a

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-cloudwatch-exporter/Chart.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 apiVersion: v1
-appVersion: "0.12.2"
+appVersion: "0.14.0"
```
Comment (monotek) [inline charts/prometheus-cloudwatch-exporter/Chart.yaml:2]: why is 0.13.0 skipped?
```suggestion
appVersion: "0.13.0"
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — CHANGES_REQUESTED** [review]
please sign your commits to make the dco check work...

**Comment by duffn** [comment]
I've signed correctly now.

File: charts/prometheus-cloudwatch-exporter/Chart.yaml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 apiVersion: v1
-appVersion: "0.12.2"
+appVersion: "0.13.0"
```
Comment (monotek) [inline charts/prometheus-cloudwatch-exporter/Chart.yaml:2]: sorry, if thought we've skipped on version in the "version:" filed.
0.14.0 would be correct if tag is 0.14.0

```suggestion
appVersion: "0.14.0"
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**